collectors/channels_collector.go

```diff
@@ -174,10 +184,45 @@ func NewChannelsCollector(lnd lndclient.LightningClient, errChan chan<- error,
 			[]string{"amount"}, nil,
 		),
 
-		lnd:         lnd,
-		primaryNode: cfg.PrimaryNode,
-		errChan:     errChan,
+		lnd:                 lnd,
+		primaryNode:         cfg.PrimaryNode,
+		closedChannelsCache: nil,
+		errChan:             errChan,
+		quit:                quitChan,
 	}
+
+	// Perform an initial refresh for the cache
+	collector.refreshClosedChannelsCache()
+
+	// Start a ticker to update the cache once per 10m
+	go func() {
+		ticker := time.NewTicker(10 * time.Minute)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				collector.refreshClosedChannelsCache()
+
+			case <-collector.quit:
+				return
+			}
+		}
+	}()
+
+	return collector
+}
+
+// refreshClosedChannelsCache acquires a mutex write lock to update
+// the closedChannelsCache.
+func (c *ChannelsCollector) refreshClosedChannelsCache() {
+	data, err := c.lnd.ClosedChannels(context.Background())
+	if err != nil {
+		c.errChan <- err
+		return
```

Comment:
I think we should return the error here and handle it outside (in the goroutine).